### PR TITLE
Onboarding: Add DIFM landing page

### PR DESCRIPTION
**Pull request by aneeshd16**
### Proposed Changes

Context: pdh1Xd-T9-p2#comment-1169

* Adds a reusable landing page for DIFM.
* Deletes the `choose-service` step which was present in the `website-design-services` flow.
* Adds a `difmStartingPoint` step to the Stepper flow. This was required to make the Skip button work correctly with the selected intent/goals on the goals step.

#### Starting from Goals step

* Go to `/start`. Select a free domain and plan and proceed to the goals step.
* Click on the DIFM intent ("Hire our experts to create your dream site").
* The next step should match the screen below.

https://user-images.githubusercontent.com/5436027/178246994-d4f3f205-b289-4686-baa6-50e02cb0e200.mp4

* The slideshow should cycle between 5 images with a fade animation.
* On smaller screens, the slideshow should not be visible.
* Click on the "Show Frequently Asked Questions" button to expand the FAQ section, it should scroll the window so that the FAQ section is in view.
* Click on the button again to hide the FAQ section.
* Clicking on "Hire a Professional" should allow you to proceed through the flow till checkout.

#### Starting from Goals step (testing Skip & Back behaviour)

* Go to `/setup/goals?siteSlug=<siteslug>`
* Click on the DIFM intent ("Hire our experts to create your dream site").
* Click on "Skip for now". It should navigate to the verticals step (if enabled) and then to the design setup step.
* Check the "back" behaviour on different steps.

#### Starting from the landing page
* Go to `/start/do-it-for-me`.
* Test both New Site and Existing Site flows.

Fixes Automattic/martech#835

### CODE REVIEW

**Comment by ddc22** [comment]
Hey @aneeshd16 should I hold off on the review for this?

**Comment by aneeshd16** [comment]
> Hey @aneeshd16 should I hold off on the review for this?

Hi @jdc91, this PR is ready for review now!

File: client/my-sites/marketing/do-it-for-me/difm-landing.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.difm-landing__price-placeholder {
```
Comment (ddc22) [inline client/my-sites/marketing/do-it-for-me/difm-landing.scss]: Is there a reason that this was moved to sass? :)

File: client/my-sites/marketing/do-it-for-me/difm-landing.tsx [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+	const translate = useTranslate();
+	const displayCost = useSelector( ( state ) => getProductDisplayCost( state, WPCOM_DIFM_LITE ) );
+	const isLoading = useSelector( isProductsListFetching );
+	const faqHeader = useRef( null );
+
+	const [ isFAQSectionOpen, setIsFAQSectionOpen ] = useState( false );
+
+	const onFAQButtonClick = () => {
+		setIsFAQSectionOpen( ! isFAQSectionOpen );
+	};
```
Comment (ddc22) [inline client/my-sites/marketing/do-it-for-me/difm-landing.tsx]: Nit. Probably can be added inline :)

File: client/my-sites/marketing/do-it-for-me/difm-landing.tsx [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+								},
+							}
+						) }
+					/>
+					<VerticalStepProgress>
+						<Step
+							index={ translate( '1' ) }
+							title={ translate( 'Submit your business information' ) }
+							description={ translate( 'Optionally provide your profiles to be found on social.' ) }
+						/>
```
Comment (ddc22) [inline client/my-sites/marketing/do-it-for-me/difm-landing.tsx:307]: Nothing wrong here but, is there a reason why this is part moved to a separate component?

**Review by ddc22 — APPROVED** [review]
Overall works great I have left some minor non critical code feedback. And some general non critical feedback 😁 . Feel free to ignore them!

Maybe we can come up with better name for difmStartingPoint because it will be visible to the user
image

Maybe we can show only one FAQ open and auto close others .
image

 If instead if you think it's better to allow multiple FAQs to be open at once maybe we can make the entire expanded FAQ clickable rather than the vague upper section of it.
image

File: client/my-sites/marketing/do-it-for-me/difm-landing.scss (hunk 1 above)
Comment (aneeshd16) [reply client/my-sites/marketing/do-it-for-me/difm-landing.scss]: Initially, I added this file to reuse some SASS variables, but it is unnecessary. I have deleted this file in e571c2f40c8cb276171f1ce662949be967578558.

**Comment by aneeshd16** [comment]
Thanks for the review @jdc91!

> Maybe we can come up with better name for difmStartingPoint because it will be visible to the user !

I chose this name following a precedent: https://github.com/Automattic/wp-calypso/blob/c88fe5386036b0554945fb62b93cdde960ada19c/client/landing/stepper/declarative-flow/site-setup-flow.ts#L40
But open to changing this to something else!

> Maybe we can show only one FAQ open and auto close others . 

I tried this in an earlier iteration but it caused the page to jump around. More: pdh1Xd-T9-p2#comment-1175

> If instead if you think it's better to allow multiple FAQs to be open at once maybe we can make the entire expanded FAQ clickable rather than the vague upper section of it. 

I agree the clickable area is not clearly defined, but my assumption was that users will click on the title or the arrow to minimize the FAQ.

File: client/my-sites/marketing/do-it-for-me/difm-landing.tsx (hunk 2 above)
Comment (aneeshd16) [reply client/my-sites/marketing/do-it-for-me/difm-landing.tsx]: I memoized it instead in 12ae758904670a66192485f67cdcc39c6fb97e08. 😉

File: client/my-sites/marketing/do-it-for-me/difm-landing.tsx (hunk 3 above)
Comment (aneeshd16) [reply client/my-sites/marketing/do-it-for-me/difm-landing.tsx:307]: This is in a separate component to abstract away the styling elements (multiple flex divs) and it also makes the parent component easier to read.

**Review by ivan-ottinger — COMMENTED** [review]
Starting from Goals step
→ ✅ Works as expected. The landing page looks and works well. The buttons and links works as expected, including the FAQ button. Also reviewed the mobile layout. The placeholder image disappears there, but looking at the CSS, this is expected.

**Starting from Goals step (testing Skip & Back behaviour)**
→ ✅ Works as expected. The "Back" link navigates to the Goals step correctly; Both "Skip for now" (the one on the top right-hand corner and the one in next to the "Hire a Professional" button) links lead to the Vertical step.

One thing I noticed is that once we click the "Hire a Professional" button and get to the `/start/website-design-services/difm-options` flow/step, there's no way to go back. But I think this is expected behavior.

**Starting from the landing page**
→ ✅ Worked well for both new and existing site; I reached the checkout page without any issues.

---

**Small note on the new Goals step layout**

Once we have the https://github.com/Automattic/wp-calypso/pull/65158 PR merged, we will need to double-check the navigation from the new layout of the Goals step to the DIFM landing page (and back) works well too. Since the navigation relies on the Intent, it should all work correctly, we just might to confirm to be sure.

The new Goals step layout will be hidden behind a feature flag initially, so there's no blocker to deploy either of the two PRs.

**Review by ddc22 — APPROVED** [review]
Nice use of animation! :)
The timing and speed of the animation looks good to me, however it would be good to get some feedback from @mikeicode and Lucas if possible.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7434465

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7434465).

**Pull request merged**